py/fiberassign/fba_tertiary_io.py

```diff
@@ -0,0 +1,631 @@
+
+# AR release is hard-coded to 8888 for tertiary programs
+release = 8888
+
+# AR default values
+default = {
+    "TIME_MJD_BEGIN": "2020-01-01T00:00:00+00:00",
+    "TIME_MJD_END": "2120-01-01T00:00:00+00:00",
+    "TOO_PRIO": "HI",  # for TOO_PRIO and SCND_TARGET
+}
+
+# AR allowed values for some of the required header keywords
+allow_hdrkeys = {
+    "OBSCONDS": ["BRIGHT", "DARK"],  # AR not allowing BACKUP
+    "SBPROF": ["ELG", "BGS", "PSF", "FLT"],
+}
+
+# AR required columns from {args.targdir}/tertiary-targets-{args.prognum}.fits
+req_keys = [
+    "TARGETID",  # AR should be encode_targetid(release=8888, brickid=PROGNUM, objid=np.arange(len(d)))
+    "RA",
+    "DEC",
+    "PMRA",
+    "PMDEC",
+    "REF_EPOCH",
+    "TERTIARY_TARGET",
+    "CHECKER",
+]
+
+# AR required keywords in the header
+req_hdrkeys = [
+    "FAPRGRM",
+    "OBSCONDS",
+    "SBPROF",
+    "GOALTIME",
+]
+
+# AR reproducibility for SUBPRIORITY
+np_rand_seed = 1234
+np.random.seed(np_rand_seed)
+
+
+def get_targdir(prognum):
+    """
+    Get the fiducial folder name for a given tertiary PROGNUM.
+
+    Args:
+        prognum: the tertiary PROGNUM (int)
+
+    Returns:
+        os.path.join(os.getenv("DESI_SURVEYOPS"), "tertiary", "{:04d}".format(prognum)) (string)
+    """
+    return os.path.join(os.getenv("DESI_SURVEYOPS"), "tertiary", "{:04d}".format(prognum))
+
+
+def get_targfn(prognum, targdir=None):
+    """
+    Get the full path to a tertiary PROGNUM input targets file.
+
+    Args:
+        prognum: the tertiary PROGNUM (int)
+        targdir (optional, defaults to get_targdir()): folder name (string)
+
+    Returns:
+        os.path.join(targdir, "tertiary-targets-{:04d}.fits".format(prognum)) (string)
+    """
+    if targdir is None:
+        targdir = get_targdir(prognum)
+    return os.path.join(targdir, "tertiary-targets-{:04d}.fits".format(prognum))
+
+
+def get_priofn(prognum, targdir=None):
+    """
+    Get the full path to a tertiary PROGNUM input priorities file.
+
+    Args:
+        prognum: the tertiary PROGNUM (int)
+        targdir (optional, defaults to get_targdir()): folder name (string)
+
+    Returns:
+        os.path.join(targdir, "tertiary-priorities-{:04d}.ecsv".format(prognum)) (string)
+    """
+    if targdir is None:
+        targdir = get_targdir(prognum)
+    return os.path.join(targdir, "tertiary-priorities-{:04d}.ecsv".format(prognum))
+
+
+def get_toofn(prognum, tileid, targdir=None):
+    """
+    Get the full path to a tertiary PROGNUM ToO target file for a given tile.
+
+    Args:
+        prognum: the tertiary PROGNUM (int)
+        tileid: the tile TILEID (int)
+        targdir (optional, defaults to get_targdir()): folder name (string)
+
+    Returns:
+        os.path.join(targdir, "ToO-{:04d}-{:06d}.ecsv".format(prognum, tileid)) (string)
+    """
+    if targdir is None:
+        targdir = get_targdir(prognum)
+    return os.path.join(targdir, "ToO-{:04d}-{:06d}.ecsv".format(prognum, tileid))
```

Comment:
if I m correct, fiberassign should work with any file name.
nevertheless, as these `"ToO-"` files have the same structure (=column content) as the "official" `ToO.ecsv` files, I d vote to keep the name starting with "ToO".
I could be convinced otherwise, though; @schlafly : do you have any preference?

Reply:
I'm fine either way, I just wanted to check everything was consistent. Let's see if Eddie cares one way or the other!

Reply:
I'm happy calling the input files provided from the outside as tertiary- files, and the processed file that comes out of this as a ToO file.

Reply:
ok great, same here, current version is fine then.